Framework/MDAlgorithms/src/HYSPECScatteredTransmissionCorrectionMD.cpp

```diff
@@ -0,0 +1,207 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+// local
+#include "MantidMDAlgorithms/HYSPECScatteredTransmissionCorrectionMD.h"
+
+// 3rd party
+#include "MantidAPI/ExperimentInfo.h"
+#include "MantidDataObjects/MDEventFactory.h"
+#include "MantidDataObjects/MDEventWorkspace.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/MultiThreaded.h"
+
+// standard
+#include <limits>
+#include <math.h>
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+namespace Mantid {
+namespace MDAlgorithms {
+
+constexpr double EMPTY_FLT() noexcept { return std::numeric_limits<float>::max() / 2; }
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(HYSPECScatteredTransmissionCorrectionMD)
+
+//---------------------------------------------------------------------------------------------------------
+
+void HYSPECScatteredTransmissionCorrectionMD::init() {
+  declareProperty(std::make_unique<WorkspaceProperty<IMDEventWorkspace>>("InputWorkspace", "", Direction::Input),
+                  "Input MDEventWorkspace. Either QSample (or QLab) frame plus DeltaE, or just |Q| plus DeltaE");
+
+  auto mustBePositive = std::make_shared<BoundedValidator<double>>();
+  mustBePositive->setLower(0.0);
```

Comment:
https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/inc/MantidKernel/BoundedValidator.h#L38 Make it exclusive